org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/JavaSearchScope.java

```diff
@@ -436,15 +444,26 @@ private boolean encloses(String enclosingPath, String path, int index) {
 		// if looking at a package, this scope encloses the given path
 		// if the given path is a direct child of the folder
 		// or if the given path path is the folder path (see bug 13919 Declaration for package not found if scope is not project)
-		if (path.startsWith(enclosingPath)
-			&& ((enclosingPath.length() == path.lastIndexOf('/'))
-				|| (enclosingPath.length() == path.length()))) {
+		if (isEnclosing(enclosingPath, path)) {
 			return true;
 		}
+		String module = this.modulePaths[index];
+		if (module != null) {
+			String enclosingPathWithModule = module + org.eclipse.jdt.core.search.IJavaSearchScope.JAR_FILE_ENTRY_SEPARATOR + enclosingPath;
```

Comment:
I'm guessing there are multiple cases here, since the pipe that we see is for jars. I'll check if the separator remains a pipe for workspace projects with modules.